### DISCUSSION TITLE
AllValuesFilter don't allow to ignore empty values

### DISCUSSION THREAD

**Discussion by nepoveda** [carltongibson/django-filter#1415] [Q&A]
I was trying to use AllValuesFilter on my CharFields, but i wanna ignore empty values but changing `null_value` did not help.

In my oponion It's happening here: https://github.com/carltongibson/django-filter/blob/79b4ea7535b5d6760afad736361dff08bfd1fda9/django_filters/filters.py#L484

so for my purpose i change this line to

`self.extra['choices'] = [(o, o) for o in qs if o]`

For more komplexity I maybe add `allow_blank` or something similar for this filter and make it optional

**Comment by carltongibson** [comment]
A `None` choice doesn’t make much sense. I’d suggest filtering `qs` rather than adding code here.

**Reply by peterhijma** [reply]
I don't think the @nepoveda meant that he wants a `None` choice. He wants to get rid of the `None` option which is added if you have any empty fields in the database!  If you use this filter value (`None`) => it seems to do nothing.
